test/TerraformTaintCommandTest.groovy

```diff
@@ -0,0 +1,77 @@
+import static org.hamcrest.Matchers.equalTo
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.times
+import static org.mockito.Mockito.verify
+
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(ResetStaticStateExtension.class)
+class TerraformTaintCommandTest {
+    @Nested
+    public class WithResource {
+        @Test
+        void defaultsToEmpty() {
+            def command = new TerraformTaintCommand()
+
+            def actualCommand = command.toString()
+            assertThat(actualCommand, equalTo("echo \"No resource set, skipping 'terraform taint'."))
```

Comment:
I'm fine with this, but looking at the source code, this isn't actually going to execute is it?  `command.toString()` is called within a guard that should ensure that the resource was set.  

```
            if (untaintCommand.resource && shouldApply()) {
                echo "Running '${untaintCommand.toString()}'. TerraformTaintPlugin is enabled."
                sh untaintCommand.toString()
            }
```

I'd be fine with taking this functionality out ("this functionality" = echo if resource is not set), and letting the command fail on its own (ie: it should fail because `terraform taint` expects a resource which will not be provided, and will return a non-zero response code and display the help command).  But I'm fine leaving it in if you think it's a better experience.

Reply:
It's true that the test code is testing code that won't execute in practice. I'm not sure where to fall on failure. I don't mind it failing by having the command blow up, it's more obvious to the user.

I'll remove the guard on having the resource set.